mock/py/mockbuild/buildroot.py

```diff
@@ -576,11 +581,19 @@ def _lock_buildroot(self, exclusive):
         try:
             fcntl.lockf(self._lock_file.fileno(), lock_type | fcntl.LOCK_NB)
         except IOError:
+            # If the lock file cannot be locked then this process does not own it.
+            # Setting it to None to signal that no lock could be acquired
+            self._lock_file = None
             raise BuildRootLocked("Build root is locked by another process.")
 
     @traceLog()
     def _unlock_buildroot(self):
         if self._lock_file:
+            # We acquired a lock. Hence we can delete the lock file.
+            # Deleting the file must happen before closing it.
+            # Otherwise other process can acquire the lock after this process
+            # closes the lock file but before it deletes it, causing a race condition.
+            os.unlink(os.path.join(os.path.abspath(self.basedir)+ ".lock"))
```

Comment:
This doesn't feel right, I'm not sure.  What if some mock process still holds a non-exclusive lock?  Another Mock process can come then and take the exclusive/write lock, and modify the chroot contents under the hands of the first process.  WDYT?